apis/logging/v1/input_receiver_types.go

```diff
@@ -2,20 +2,37 @@ package v1
 
 // NOTE: The Enum validation on ReceiverSpec.Type must be updated if the list of types changes.
 
+// Receiver type constants, must match JSON tags of OutputTypeSpec fields.
+const (
+	ReceiverTypeHttp   = "http"
+	ReceiverTypeSyslog = "syslog"
+
+	FormatKubeAPIAudit = "kubeAPIAudit" // Log events in k8s list format, e.g. API audit log events.
+)
+
 // ReceiverSpec is a union of input Receiver types.
 //
 // The fields of this struct define the set of known Receiver types.
 type ReceiverSpec struct {
-	HTTP *HTTPReceiver `json:"http,omitempty"`
+
+	// Type of Receiver plugin.
+	//
+	// +kubebuilder:validation:Enum:=http;syslog
+	// +required
+	Type string `json:"type"`
+
+	// The ReceiverTypeSpec that handles particular parameters
+	*ReceiverTypeSpec `json:",inline"`
 }
 
-const (
-	FormatKubeAPIAudit = "kubeAPIAudit" // Log events in k8s list format, e.g. API audit log events.
-)
+type ReceiverTypeSpec struct {
+	HTTP   *HTTPReceiver   `json:"http,omitempty"`
+	Syslog *SyslogReceiver `json:"syslog,omitempty"`
+}
 
 // HTTPReceiver receives encoded logs as a HTTP endpoint.
 type HTTPReceiver struct {
-	// Port the Service and the HTTP listener listen on.
+	// Port the Receiver listens on.
 	// +kubebuilder:default:=8443
```

Comment:
We should mention here and for syslog that we only consider ports [1024, 65535] to be valid
https://book.kubebuilder.io/reference/markers/crd-validation says there's `+kubebuilder:validation:Minimum and Maximum`